flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/threadinfo/JobVertexThreadInfoTracker.java

```diff
@@ -204,31 +206,33 @@ private void triggerThreadInfoSampleInternal(
         }
     }
 
-    private Map<Set<ExecutionAttemptID>, CompletableFuture<TaskExecutorThreadInfoGateway>>
+    private Map<ImmutableSet<ExecutionAttemptID>, CompletableFuture<TaskExecutorThreadInfoGateway>>
```

Comment:
I think it makes it much clearer to use `ImmutableSet` for keys. So I'd keep the change.